src/resolve-dependency.ts

```diff
@@ -284,60 +214,33 @@ function resolveExportsImports(
     }
   }
 
-  for (const match of Object.keys(matchObj).sort(
-    (a, b) => b.length - a.length
-  )) {
-    if (!match.endsWith("/")) continue;
+  for (const match of Object.keys(matchObj).sort((a, b) => b.length - a.length)) {
```

Comment:
This sort should be updated to use the order defined by `PATTERN_KEY_COMPARE` in https://nodejs.org/dist/latest-v20.x/docs/api/esm.html#resolution-and-loading-algorithm.